drivers/display/display_st7701.c

```diff
@@ -0,0 +1,459 @@
+
+	const uint8_t _E6[] = { 0xE6, 0x00, 0x00, 0x33, 0x33 };
+	const uint8_t _E7[] = { 0xE7, 0x44, 0x44 };
+
+	st7701_generic_write(dev, _E6, sizeof(_E6));
+	st7701_generic_write(dev, _E7, sizeof(_E7));
+
+	const uint8_t _E8[] = { 0xE8, 0x0D, 0x60, 0xA0, 0xA0, 0x0F,
+							0x60, 0xA0, 0xA0, 0x09, 0x60, 0xA0,
+							0xA0, 0x0B, 0x60, 0xA0, 0xA0 };
+
+	st7701_generic_write(dev, _E8, sizeof(_E8));
+
+	const uint8_t _EB[] = { 0xEB, 0x02, 0x01, 0xE4, 0xE4, 0x44, 0x00, 0x40 };
+	const uint8_t _EC[] = { 0xEC, 0x02, 0x01 };
+
+	st7701_generic_write(dev, _EB, sizeof(_EB));
+	st7701_generic_write(dev, _EC, sizeof(_EC));
+
+	const uint8_t _ED[17] = { 0xED, 0xAB, 0x89, 0x76, 0x54, 0x01,
+							0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF,
+							0x10, 0x45, 0x67, 0x98, 0xBA };
+
+	st7701_generic_write(dev, _ED, sizeof(_ED));
+
+	/* Bank1 setting */
+	const uint8_t _FF2[] = { DSI_CMD2BKX_SEL, 0x77, 0x01, 0x00, 0x00, DSI_CMD2BKX_SEL_NONE };
+
+	st7701_generic_write(dev, _FF2, sizeof(_FF2));
+
+	/* Exit sleep mode */
+	ret = st7701_dcs_write(dev, MIPI_DCS_EXIT_SLEEP_MODE, NULL, 0);
+	if (ret < 0) {
+		return ret;
+	}
+
+	k_msleep(50);
+
+	/* Set pixel color format */
+	switch (data->dsi_pixel_format) {
+	case MIPI_DSI_PIXFMT_RGB565:
+		buf[0] = MIPI_DCS_PIXEL_FORMAT_16BIT;
+		break;
+	case MIPI_DSI_PIXFMT_RGB888:
+		buf[0] = MIPI_DCS_PIXEL_FORMAT_24BIT;
+		break;
+	default:
+		LOG_ERR("Unsupported pixel format 0x%x!", data->dsi_pixel_format);
+		return -ENOTSUP;
+	}
+
+	ret = st7701_dcs_write(dev, MIPI_DCS_SET_PIXEL_FORMAT, buf, 1);
+	if (ret < 0) {
+		return ret;
+	}
+
+	buf[0] = 0x00;
+	buf[1] = 0x00;
+	sys_put_be16(data->xres, (uint8_t *)&buf[2]);
+	ret = st7701_dcs_write(dev, MIPI_DCS_SET_COLUMN_ADDRESS, buf, 4);
+	if (ret < 0) {
+		return ret;
+	}
+
+	buf[0] = 0x00;
+	buf[1] = 0x00;
+	sys_put_be16(data->yres, (uint8_t *)&buf[2]);
+	ret = st7701_dcs_write(dev, MIPI_DCS_SET_PAGE_ADDRESS, buf, 4);
+	if (ret < 0) {
+		return ret;
+	}
+
+	/* Backlight control */
+	buf[0] = ST7701_WRCTRLD_BCTRL | ST7701_WRCTRLD_DD | ST7701_WRCTRLD_BL;
+	ret = st7701_dcs_write(dev, MIPI_DCS_WRITE_CONTROL_DISPLAY, buf, 1);
+	if (ret < 0) {
+		return ret;
+	}
+
+	/* Adaptive brightness control */
+	buf[0] = ST7701_WRCABC_UI;
+	ret = st7701_dcs_write(dev, MIPI_DCS_WRITE_POWER_SAVE, buf, 1);
+	if (ret < 0) {
+		return ret;
+	}
+
+	/* Adaptive brightness control minimum brightness */
+	buf[0] = 0xFF;
+	ret = st7701_dcs_write(dev, MIPI_DCS_SET_CABC_MIN_BRIGHTNESS, buf, 1);
+	if (ret < 0) {
+		return ret;
+	}
+
+	/* Brightness */
+	buf[0] = 0xFF;
+	ret = st7701_dcs_write(dev, MIPI_DCS_SET_DISPLAY_BRIGHTNESS, buf, 1);
+	if (ret < 0) {
+		return ret;
+	}
+
+	/* Display On */
+	ret = st7701_dcs_write(dev, MIPI_DCS_SET_DISPLAY_ON, NULL, 0);
+	if (ret < 0) {
+		return ret;
+	}
+
+	return 0;
+}
+
+static int st7701_blanking_on(const struct device *dev)
+{
+	const struct st7701_config *cfg = dev->config;
+	int ret;
+
+	if (cfg->backlight.port != NULL) {
+		ret = gpio_pin_set_dt(&cfg->backlight, 0);
```

Comment:
Setting the backlight to `ON` we typically set it to a logical `1`. You can invert the polarity in the device tree.

Reply:
This is meant to disable it, see this driver for an example: https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/display/display_ili9806e_dsi.c#L288

Reply:
My bad.